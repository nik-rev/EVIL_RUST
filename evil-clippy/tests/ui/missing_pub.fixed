#![warn(clippy::missing_pub)]

pub struct StructA;
pub struct StructB;
//~^ missing_pub

pub fn function_a() {}
//~^ missing_pub
pub fn function_b() {}

pub trait TraitA {}
//~^ missing_pub
pub trait TraitB {}

pub mod module_a {}
//~^ missing_pub
pub  mod module_b {}
//~^ missing_pub
pub mod module_c {}

pub static STATIC_A: () = ();
//~^ missing_pub
pub static STATIC_B: () = ();

pub const CONST_A: () = ();
//~^ missing_pub
pub const CONST_B: () = ();

pub fn main() {}
//~^ missing_pub
