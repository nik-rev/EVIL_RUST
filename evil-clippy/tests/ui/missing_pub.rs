#![warn(clippy::reference_used)]

fn main() {
    let x = 10;
    let y = &x;
    //~^ reference_used

    let z = &mut 20;
    //~^ reference_used

    let (a, ref b) = (1, 2);
    //~^ reference_used

    let c = Some(3);
    if let Some(ref d) = c {
        //~^ reference_used
        println!("{}", d);
    }

    let s = String::from("hello");
    let ref_s = &s;
    //~^ reference_used

    // Function call with reference
    takes_ref(&x);
    //~^ reference_used

    // Mutable reference passed
    takes_mut_ref(&mut 42);
    //~^ reference_used

    // Raw pointers are allowed
    let raw_const: *const i32 = &x;
    let raw_mut: *mut i32 = &mut 100;
    unsafe {
        println!("{}", *raw_const);
        *raw_mut = 200;
    }

    // Closure capturing by reference
    let closure = |val: &i32| println!("{}", val);
    //~^ reference_used
    closure(&x);
    //~^ reference_used

    // Closure with mutable reference
    let mut val = 5;
    let closure_mut = |v: &mut i32| *v += 1; //~^ reference_used
    closure_mut(&mut val); //~^ reference_used

    // Struct with reference field (should trigger)
    struct RefStruct<'a> {
        r: &'a i32, //~^ reference_used
    }

    // Trait method using reference
    trait RefTrait {
        fn ref_method(&self); //~^ reference_used
    }

    struct MyStruct;
    impl RefTrait for MyStruct {
        fn ref_method(&self) {
            //~^ reference_used
            println!("Hello from ref method");
        }
    }

    let my_struct = MyStruct;
    my_struct.ref_method(); //~^ reference_used

    // Array of references
    let arr: [&i32; 2] = [&x, &42]; //~^ reference_used

    // Tuple with references
    let tup: (&i32, &mut i32) = (&x, &mut val); //~^ reference_used

    // Function pointer with reference param
    let _f: fn(&i32) = takes_ref; //~^ reference_used
}

// Function taking reference
fn takes_ref(val: &i32) {
    //~^ reference_used
    println!("{}", val);
}

// Function taking mutable reference
fn takes_mut_ref(val: &mut i32) {
    //~^ reference_used
    *val += 1;
}
