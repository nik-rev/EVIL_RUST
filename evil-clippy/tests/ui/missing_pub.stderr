error: item is not public
  --> tests/ui/missing_pub.rs:4:1
   |
LL | struct StructB;
   | ^ help: make this item public: `pub`
   |
   = note: `-D clippy::missing-pub` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::missing_pub)]`

error: item is not public
  --> tests/ui/missing_pub.rs:7:1
   |
LL | fn function_a() {}
   | ^ help: make this item public: `pub`

error: item is not public
  --> tests/ui/missing_pub.rs:11:1
   |
LL | trait TraitA {}
   | ^ help: make this item public: `pub`

error: item is not public
  --> tests/ui/missing_pub.rs:15:1
   |
LL | mod module_a {}
   | ^ help: make this item public: `pub`

error: item is not public
  --> tests/ui/missing_pub.rs:17:1
   |
LL | pub(crate) mod module_b {}
   | ^^^^^^^^^^ help: make this item public: `pub`

error: item is not public
  --> tests/ui/missing_pub.rs:21:1
   |
LL | static STATIC_A: () = ();
   | ^ help: make this item public: `pub`

error: item is not public
  --> tests/ui/missing_pub.rs:25:1
   |
LL | const CONST_A: () = ();
   | ^ help: make this item public: `pub`

error: item is not public
  --> tests/ui/missing_pub.rs:29:1
   |
LL | fn main() {}
   | ^ help: make this item public: `pub`

error: aborting due to 8 previous errors

